lib/pos_blockchain.go

```diff
@@ -18,45 +18,116 @@ import (
+		if addToBlockIndexErr := bc.storeValidateFailedBlockInBlockIndex(desoBlock); addToBlockIndexErr != nil {
+			return false, false, nil, errors.Wrapf(err, "processBlockPoS: Problem adding validate failed block to block index: %v", addToBlockIndexErr)
+		}
+		return false, false, nil, errors.Wrap(err, "processBlockPoS: block's parent failed validation: ")
+	}
+	if !parentBlockNode.IsValidated() {
+		blk, err := GetBlock(parentBlockNode.Hash, bc.db, bc.snapshot)
 		if err != nil {
-			return false, true, missingBlockHashes, err
+			return false, false, nil, errors.Wrapf(err, "processBlockPoS: Problem getting parent block %v", parentBlockNode.Hash)
 		}
-		// ProcessOrphanBlock validates the block and adds it to the orphan list.
-		// TODO: update _validateOrphanBlock to perform additional validation required.
-		if err = bc.ProcessOrphanBlock(desoBlock, blockHash); err != nil {
-			return false, true, missingBlockHashes, errors.Wrap(err, "processBlockPoS: Problem processing orphan block: ")
+		_, _, _, err = bc.processBlockPoS(blk, currentView, verifySignatures)
+		if err != nil {
+			return false, false, nil, errors.Wrapf(err, "processBlockPoS: Problem processing parent block %v", parentBlockNode.Hash)
 		}
-		return false, true, missingBlockHashes, nil
 	}
 
-	// 2. Start with all sanity checks of the block.
+	// At this point, we know that all the ancestors of this block have been successfully
+	// validated. That means we can now proceed with attempting to validate THIS block.
+
+	// Do all the sanity checks on the block.
 	// TODO: Check if err is for view > latest committed block view and <= latest uncommitted block.
 	// If so, we need to perform the rest of the validations and then add to our block index.
 	if err = bc.validateDeSoBlockPoS(desoBlock); err != nil {
+		// TODO: If validations fail, do we want to even bother storing the block? Are there any validations that
+		// wouldn't result in us wanting to mark the block as StatusBlockValidateFailed?
+		// If validations fail, we can update the block index with the block status ValidateFailed
+		if addToBlockIndexErr := bc.storeValidateFailedBlockInBlockIndex(desoBlock); addToBlockIndexErr != nil {
+			return false, false, nil, errors.Wrapf(err, "processBlockPoS: Problem adding validate failed block to block index: %v", addToBlockIndexErr)
+		}
 		return false, false, nil, errors.Wrap(err, "processBlockPoS: block validation failed: ")
 	}
 
+	// We know the utxoView will be valid because we check that all ancestor blocks have been validated.
 	utxoView, err := bc.getUtxoViewAtBlockHash(*desoBlock.Header.PrevBlockHash)
 	if err != nil {
 		return false, false, nil, errors.Wrap(err, "processBlockPoS: Problem initializing UtxoView: ")
 	}
+	// Connect this block to the UtxoView.
+	txHashes := collections.Transform(desoBlock.Txns, func(txn *MsgDeSoTxn) *BlockHash {
+		return txn.Hash()
+	})
+	_, err = utxoView.ConnectBlock(desoBlock, txHashes, true, bc.eventManager, desoBlock.Header.Height)
```

Comment:
Do we really want to pass through the bc.eventManager here? I'm not sure what the downstream implications are for event listeners that have subscribed to it

Reply:
You're right. we shouldn't since this is a throw away view.